### DISCUSSION TITLE
ROCDebugger and Seergdb.

### DISCUSSION THREAD

**Discussion by epasveer** [ROCm/ROCm#2243] [General]
Hi All,

Sorry for the, potentially, non-relevant post in this group.

I'm the developer for a gui frontend to gdb called Seer.

https://github.com/epasveer/seer

It uses gdb-mi. I came across ROCDebugger that seems to be a custom version of gdb that is for AMD, and it provides a gdb-mi interface as well.

I wonder if some could try ROCDebugger and Seer together. Seer has a config dialog where you can specify an alternate gdb program.

Does ROCDebugger provide any extra features over the standard gdb?   Perhaps I can add features to Seer to support them.

Thanks for your attention.

--Ernie

**Comment by zichguan-amd** [comment]
Hi @epasveer, thanks for your interest in collaborating with the ROCm stack. ROCgdb extends standard gdb to enable heterogenous debugging on both CPU and AMDGPU. It has added GPU concepts of agents (GPUs), queues, dispatches, and lanes. Anything that works with gdb is still compatible with ROCgdb, so all the standard gdb features that Seer supports should work with ROCgdb out of the box.

You can find the source code [here](https://github.com/ROCm/ROCgdb) and more details in our [documentation](https://rocm.docs.amd.com/projects/ROCgdb/en/latest/). You may also find it easier to check the differences between ROCgdb and gdb in our manual with `git diff master..amd-staging -- gdb/doc/gdb.texinfo` in [ROCgdb repo](https://github.com/ROCm/ROCgdb).

We are happy to help if you have further questions. Feel free to comment here or post issues directly in [ROCgdb repo](https://github.com/ROCm/ROCgdb).

**Reply by epasveer** [reply]
Thanks for the reply.   I'm interested if ROCgdb provides additional gdb "mi" calls.  I will check the code.

**Reply by zichguan-amd** [reply]
You can look for 'Heterogeneous' in section titles to see what pertains to GPU extensions, maybe these sections could help https://rocm.docs.amd.com/projects/ROCgdb/en/latest/ROCgdb/gdb/doc/gdb/GDB_002fMI-Heterogeneous-Agent-Information.html, https://rocm.docs.amd.com/projects/ROCgdb/en/latest/ROCgdb/gdb/doc/gdb/GDB_002fMI-Heterogeneous-Queue-Information.html, https://rocm.docs.amd.com/projects/ROCgdb/en/latest/ROCgdb/gdb/doc/gdb/GDB_002fMI-Heterogeneous-Dispatch-Information.html, and https://rocm.docs.amd.com/projects/ROCgdb/en/latest/ROCgdb/gdb/doc/gdb/GDB_002fMI-Heterogeneous-Debugging-Commands.html

**Reply by epasveer** [reply]
Thanks!  They do help!
